engine-core/src/main/scala/org/scalarules/utils/FactMacros.scala

```diff
@@ -6,37 +6,117 @@ import scala.annotation.compileTimeOnly
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox._
 
+/**
+  * `FactMacros` contains two compile-time macros which allow the easy creation of a `Fact`, without having to explicitly repeat
+  * the name of the `Fact` as an argument. This makes it easier to define Facts for use in a DSL, as the name of the val to
+  * which it is assigned, becomes the name of the `Fact` itself.
+  *
+  * Without these macros, you would need to create `Fact`s as follows:
+  *
+  * `val MyFact = new SingularFact[Int]("MyFact")`
+  *
+  * This redundancy of the `Fact`'s name is not just a nuisance to type, it actually causes trouble when renaming the `Fact`,
+  * or trying to copy its declaration. In both situations, you *must* also alter the name parameter manually. The compiler
+  * and IDE tooling cannot help you track these errors down and so you are bound to find out about them at runtime, with
+  * potentially uninformative error messages.
+  *
+  * If instead, you use these macros, these troubles will be taken away. Define your `Fact`s like this and have the IDE and
+  * compiler warn you about duplicate names etc:
+  *
+  * `val MyFact = FactMacros.defineFact[Int]("Description")`
+  *
+  */
 object FactMacros {
 
+  // Macro definitions
+
+  /**
+    * Defines a `Fact`, using the name of the `val` it is assigned to as the name of the `Fact`. Note: the value of this
+    * macro *must* be assigned to a `val`, otherwise a compiler error will be raised.
+    *
+    * @tparam A the value type of the resulting `Fact`.
+    * @return a `SingularFact` initialized with the name of the val declaration.
+    */
+  def defineFact[A](): SingularFact[A] = macro FactMacros.defineFactMacroImpl[A]
```

Comment:
Sorry, I missed the fact that we were looking at `FactMacros`. 

I initially added these to `FactMacros` to be one unit for the macros, both definitions and implementations. Here they sort of 'naturally' belong together. In some tests I use `FactMacros.defineFact` to create facts without using a `Glossary`. The `Glossary` 'duplicates' these methods so it can offer the same functionality without needing to prefix the macro calls with `FactMacros`.

We could remove the ones in `FactMacros`, since the intended use is through extending the `Glossary` class.